### ISSUE TITLE
Launching a browser doesn't switch to that application

### ISSUE THREAD

**Issue by kputschko** [complexlogic/flex-launcher#2]
Hey there, I'm loving this launcher.  I'm trying to set up a button to launch a browser, and the following command successfully launches Edge, but Flex-Launcher still maintains focus after the browser is launched.  Any ideas why this this happening?

```
Entry1=Browser;.\assets\icons\system.png; "C:\ProgramData\Microsoft\Windows\Start Menu\Programs\Microsoft Edge.lnk"
```

**Comment by complexlogic** [comment]
Flex Launcher assumes that the launched executable won't return until the user has closed the application. The problem is that the Edge executable doesn't behave the way most programs do. The msedge.exe is really just a launcher. It works by forking a new process for the browser window, and then immediately returns. Chrome and Firefox on Windows also have the same behavior. So when you launch the Edge executable, it will successfully create a browser window, but Flex Launcher will re-grab the window focus almost immediately after, mistakenly thinking that Edge has closed.

As a workaround, you can use the following script to block until Edge has closed. Change the path to the msedge.exe if necessary.
```
@echo off
start "msedge" "C:\Program Files (x86)\Microsoft\Edge\Application\msedge.exe"

:loop
tasklist /FI "ImageName eq msedge.exe"  | find /i "msedge" > nul || goto :done
timeout /t 2 >nul
goto :loop

:done
```
It works by checking every 2 seconds to see if there are any Edge processes running, and it won't return until there are none.

Paste it into a .bat file, and then you can structure your menu entry like so:
```
Entry1=Browser;.\assets\icons\system.png;"C:\path\to\scripts\launch_edge.bat"
```
Eventually, I will incorporate similar logic into Flex Launcher itself, but you may use this workaround for the time being.

**Comment by complexlogic** [comment]
This issue is fixed in the latest release, [v1.3](https://github.com/complexlogic/flex-launcher/releases/tag/v1.3). The logic from the above script is now incorporated directly into the launcher. Upon program launch, the launcher will check the image name of the resulting process, and, if it matches ```chrome.exe```, ```msedge.exe```, or ```firefox.exe```, then it will wait until all processes of the same name have terminated before returning to the launcher screen. Launching a web browser via a shortcut to it's .exe is allowable.
